horovod/tensorflow/__init__.py

```diff
@@ -34,6 +34,7 @@
 from horovod.tensorflow.mpi_ops import mpi_threads_supported, mpi_enabled, mpi_built
```

Comment:
We might want to add the ability to specify a process set to the `DistributedOptimizer` and `DistributedGradientTape` wrappers.

Reply:
Definitely. How should the API look like?

With TF1-style legacy optimizers one currently makes calls like:
```
avg_grad_var_pairs = opt.compute_gradients([(grad0, var0), (grad1, var1), ...])
```
Should we extend this to support something like the following if we want to have grad1 averaged over a non-global process set?
```
avg_grad_var_tuples = opt.compute_gradients([(grad0, var0), (grad1, var1, process_set_1), ...]))
```
Maybe it's not a good idea to clandestinely change the signature of the base class method.

Alternatively we could support handing in a callback `get_process_set_for_variable(var)` or even a dictionary `{var: process_set, ...}` listing all variables that should live in non-global process sets.

Reply:
I was thinking of something even more simple than what you are proposing here: just adding a process set argument to the `DistributedOptimizer`/`DistributedGradientTape` constructors to limit the action of those wrappers to a provided process set. Do you think there are common scenarios where a user would want/need to have per gradient control over the process set within the same optimizer?

Reply:
Ah, excellent! Most people will indeed be able to build separate optimizers for their special variables that live in specific process sets. It's a very good idea to limit the extra complexity in `DistributedOptimizer`. Otherwise users can always wrap / subclass / hack it themselves.

I am going to add such an argument to the constructors once we have the nicer Python API with `ProcessSet` objects.

Reply:
done, and fixed some details on the way